api/controllers/console/app/conversation.py

```diff
@@ -123,6 +128,144 @@ def get(self, app_model):
 
         return conversations
 
+    @setup_required
+    @login_required
+    @account_initialization_required
+    @get_app_model(mode=AppMode.COMPLETION)
+    def delete(self, app_model):
+        """Clear completion conversations and related data including files"""
+        if not current_user.is_editor:
+            logger.warning(
+                "Unauthorized deletion attempt: user %s tried to delete completion conversations for app %s",
+                current_user.id,
+                app_model.id,
+            )
+            raise Forbidden()
+
+        parser = reqparse.RequestParser()
+        parser.add_argument("conversation_ids", type=list, location="json")
+        args = parser.parse_args()
+
+        # If specific conversation IDs provided, delete only those; otherwise delete all
+        if args["conversation_ids"]:
```

Comment:
Sorry for the late reply. :( 
The feature LGTM, however, you should move these logic into the service layer in order to maintain the consistency.